packages/amplify-migration-codegen-e2e/src/assertions.ts

```diff
@@ -5,50 +5,108 @@ import {
+  const { graphqlApi } = await getAppSyncApi(gen1GraphqlApiId, gen1Region);
 
   expect(gen1GraphqlApiId).toBeDefined();
   expect(GraphQLAPIEndpointOutput).toBeDefined();
   expect(GraphQLAPIKeyOutput).toBeDefined();
 
   expect(graphqlApi).toBeDefined();
   expect(graphqlApi?.apiId).toEqual(gen1GraphqlApiId);
+  return { gen1GraphqlApiId };
+}
+
+export async function assertDefaultGen1Setup(projRoot: string) {
+  const { gen1Meta, gen1Region } = await getMeta(projRoot);
+  const { gen1UserPoolId } = await assertUserPool(gen1Meta, gen1Region);
+  const { gen1FunctionName } = await assertFunction(gen1Meta, gen1Region);
+  const { gen1BucketName } = await assertStorage(gen1Meta, gen1Region);
+  const { gen1GraphqlApiId } = await assertAPI(gen1Meta, gen1Region);
+
   return { gen1UserPoolId, gen1FunctionName, gen1BucketName, gen1GraphqlApiId, gen1Region };
 }
 
+export async function assertAuthWithMaxOptionsGen1Setup(projRoot: string) {
+  const { gen1Meta, gen1Region } = await getMeta(projRoot);
+  const { gen1UserPoolId } = await assertUserPool(gen1Meta, gen1Region);
+  const { gen1FunctionName } = await assertFunction(gen1Meta, gen1Region);
+
+  const {
+    UserPoolId: userPoolId,
+    AppClientIDWeb: appClientIdWeb,
+    AppClientID: appClientId,
+  } = Object.keys(gen1Meta.auth).map((key) => gen1Meta.auth[key])[0].output;
+  const clientIds = [appClientIdWeb, appClientId];
+  const clients = await getUserPoolClients(userPoolId, clientIds, gen1Region);
+  expect(clients).toHaveLength(2);
+
+  const { userPoolGroups } = gen1Meta.auth;
+  expect(userPoolGroups.service).toEqual('Cognito-UserPool-Groups');
+  expect(userPoolGroups.providerPlugin).toEqual('awscloudformation');
+  expect(userPoolGroups.dependsOn.length).toBe(1);
+  expect(userPoolGroups.dependsOn[0].category).toBe('auth');
+  expect(userPoolGroups.dependsOn[0].attributes.length).toBe(4);
+  expect(userPoolGroups.dependsOn[0].attributes).toContain('UserPoolId');
+  expect(userPoolGroups.dependsOn[0].attributes).toContain('AppClientIDWeb');
+  expect(userPoolGroups.dependsOn[0].attributes).toContain('AppClientID');
+  expect(userPoolGroups.dependsOn[0].attributes).toContain('IdentityPoolId');
+
+  return { gen1UserPoolId, gen1FunctionName, gen1Region };
+}
+
+export async function assertStorageWithMaxOptionsGen1Setup(projRoot: string) {
+  const { gen1Meta, gen1Region } = await getMeta(projRoot);
+  const { gen1BucketName } = await assertStorage(gen1Meta, gen1Region);
+  return { gen1BucketName, gen1Region };
+}
+
 export async function assertUserPoolResource(projRoot: string, gen1UserPoolId: string, gen1Region: string) {
   const gen1Resource = await getResourceDetails('AWS::Cognito::UserPool', gen1UserPoolId, gen1Region);
-  removeProperties(gen1Resource, ['ProviderURL', 'ProviderName', 'UserPoolId', 'Arn']);
+  console.log(gen1Resource);
+  removeProperties(gen1Resource, ['ProviderURL', 'ProviderName', 'UserPoolId', 'Arn', 'LambdaConfig.PostConfirmation']);
```

Comment:
`LambdaConfig.PostConfirmation` <-- just want to confirm it is a function name right?

Reply:
Yes that is the function associated with lambda trigger which is added

Reply:
The arn is different for these functions in gen 1 and gen 2. Maybe I could assert if both strings are present in gen1 and gen2 stack properties before removing.